src/Microsoft.SqlTools.Hosting.v2/Utility/Logger.cs

```diff
@@ -5,86 +5,152 @@
+                TraceSource.Switch = new SourceSwitch(TraceSource.Name, value.ToString());
+                // configure the listener level filter
+                tracingLevel = value;
+                Listener.Filter = new EventTypeFilter(tracingLevel);
             }
         }
-        
+
         /// <summary>
-        /// Initializes the Logger for the current session.
+        /// Initializes the Logger for the current process.
         /// </summary>
-        /// <param name="logFilePath">
-        /// Optional. Specifies the path at which log messages will be written.
-        /// </param>
-        /// <param name="minimumLogLevel">
+        /// <param name="tracingLevel">
         /// Optional. Specifies the minimum log message level to write to the log file.
         /// </param>
-        public void Initialize(
-            string logFilePath = "sqltools",
-            LogLevel minimumLogLevel = LogLevel.Normal,
-            bool isEnabled = true)
+        /// <param name="logFilePath">
+        /// Optional. Specifies the log name prefix for the log file name at which log messages will be written.
+        /// <param name="traceSource">
+        /// Optional. Specifies the tracesource name.
+        /// </param>
+        public static void Initialize(
+            SourceLevels tracingLevel = defaultTracingLevel,
+            string logFilePath = null,
+            string traceSource = defaultTraceSrouce)
         {
-            Logger.isEnabled = isEnabled;
-            
-            // return if the logger is not enabled or already initialized
-            if (!Logger.isEnabled || Logger.isInitialized)
-            {
-                return;
-            }
+            Logger.tracingLevel = tracingLevel;
+            TraceSource = new TraceSource(traceSource, Logger.tracingLevel);
+            if (string.IsNullOrWhiteSpace(logFilePath))
+                logFilePath = GenerateLogFilePath(traceSource);
+            LogFileFullPath = logFilePath;
+            Write(TraceEventType.Information, $"Initialized the {traceSource} logger");
+        }
 
-            Logger.isInitialized = true;
+        /// <summary>
+        /// Initializes the Logger for the current process.
+        /// </summary>
+        /// </param>
+        /// <param name="tracingLevel">
+        /// Optional. Specifies the minimum log message level to write to the log file.
+        /// </param>
+        /// <param name="logFilePath">
+        /// Optional. Specifies the log name prefix for the log file name at which log messages will be written.
+        /// <param name="traceSource">
+        /// Optional. Specifies the tracesource name.
+        /// </param>
+        public static void Initialize(string tracingLevel, string logFilePath = null, string traceSource = defaultTraceSrouce)
+            => Initialize(Enum.TryParse<SourceLevels>(tracingLevel, out SourceLevels sourceTracingLevel)
```

Comment:
Minor: why use `=>` when this looks like a regular method? Is there a benefit vs 

```
{
Initialize(Enum.TryParse<SourceLevels>(tracingLevel, out SourceLevels sourceTracingLevel)
                    ? sourceTracingLevel
                    : defaultTracingLevel
                , logFilePath
                , traceSource);
}
```